x-pack/plugins/cases/server/saved_object_types/migrations/utils.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { noneConnectorId } from '../../../common';
+import { SavedObjectReference } from '../../../../../../src/core/server';
+import { ACTION_SAVED_OBJECT_TYPE } from '../../../../actions/server';
+import { getNoneCaseConnector } from '../../common';
+import { connectorIdReferenceName } from '../../services';
+
+export const transformConnectorIdToReference = (connector?: {
+  id?: string;
+}): { transformedConnector: object; references: SavedObjectReference[] } => {
+  const { id: connectorId, ...restConnector } = connector ?? {};
+
+  const references =
+    connectorId && connectorId !== noneConnectorId
+      ? [
+          {
+            id: connectorId,
+            type: ACTION_SAVED_OBJECT_TYPE,
+
+            name: connectorIdReferenceName,
+          },
+        ]
+      : [];
```

Comment:
nit: this part is almost identical in `transformPushConnectorIdToReference`. Maybe we could extract it in a function like `createConnectorReferences`.